llvm/lib/Target/AMDGPU/AMDGPUIGroupLP.cpp

```diff
@@ -2391,6 +2391,56 @@ bool SchedGroup::canAddMI(const MachineInstr &MI) const {
   if (MI.isMetaInstruction())
     Result = false;
 
+  else if (MI.isInlineAsm()) {
+    const SIRegisterInfo &TRI = TII->getRegisterInfo();
+    auto &MRI = MI.getParent()->getParent()->getRegInfo();
+    bool SGPR_used = false, VGPR_used = false, VMFMA_used = false,
+         VReg32_used = false, MayLoad = MI.mayLoad(), MayStore = MI.mayStore();
```

Comment:
I doubt mayLoad and mayStore are doing anything here. I don't believe they have the inline asm special case (like they do for hasSideEffects). Plus I'm not sure how users would wire that up